modules/irsa/main.tf

```diff
@@ -13,6 +13,22 @@ resource "kubernetes_namespace_v1" "irsa" {
   }
 }
 
+resource "kubernetes_secret_v1" "irsa" {
+  count = var.create_kubernetes_service_account ? 1 : 0
+  metadata {
+    name        = format("%s-token-secret", var.kubernetes_service_account)
+    namespace   = try(kubernetes_namespace_v1.irsa[0].metadata[0].name, var.kubernetes_namespace)
+    annotations = {
+      "kubernetes.io/service-account.name"      = var.kubernetes_service_account
+      "kubernetes.io/service-account.namespace" = try(kubernetes_namespace_v1.irsa[0].metadata[0].name, var.kubernetes_namespace)
+    }
+  }
+
+  type = "kubernetes.io/service-account-token"
+
+  depends_on = [kubernetes_service_account_v1.irsa]
```

Comment:
Is this depends_on required? I'm going to run another test taking your code snippet here with our e2e tests , removing this depends on and keeping `automount_service_account_token` to true.